src/ecanvasser/services/ecanvasser.service.ts

```diff
@@ -453,4 +486,252 @@ export class EcanvasserService extends createPrismaBase(MODELS.Ecanvasser) {
   private async sleep(ms: number) {
     await new Promise((resolve) => setTimeout(resolve, ms))
   }
+
+  async createSurvey(campaignId: number, createSurveyDto: CreateSurveyDto) {
+    const ecanvasser = await this.findByCampaignId(campaignId)
+    if (!ecanvasser) {
+      throw new NotFoundException('Ecanvasser integration not found')
+    }
+
+    const payload = {
+      name: createSurveyDto.name,
+      description: createSurveyDto.description,
+      requires_signature: createSurveyDto.requiresSignature,
+      status: createSurveyDto.status,
+      team_id: createSurveyDto.teamId,
+    } as ApiEcanvasserSurvey
+
+    try {
+      const response = await this.fetchFromApi<ApiEcanvasserSurvey>(
+        '/survey',
+        ecanvasser.apiKey,
+        {
+          method: 'POST',
+          data: payload,
+        },
+      )
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to create survey', error)
+      throw error
+    }
+  }
+
+  async findSurveys(campaignId: number) {
+    const ecanvasser = await this.findByCampaignId(campaignId)
+    if (!ecanvasser) {
+      throw new NotFoundException('Ecanvasser integration not found')
+    }
+
+    try {
+      const response = await this.fetchFromApi<ApiEcanvasserSurvey>(
+        '/survey',
+        ecanvasser.apiKey,
+      )
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to fetch surveys', error)
+      throw error
+    }
+  }
+
+  async createSurveyQuestion(
+    campaignId: number,
+    surveyId: number,
+    createQuestionDto: CreateSurveyQuestionDto,
+  ) {
+    const ecanvasser = await this.findByCampaignId(campaignId)
+    if (!ecanvasser) {
+      throw new NotFoundException('Ecanvasser integration not found')
+    }
+    const payload = {
+      survey_id: surveyId,
+      name: createQuestionDto.name,
+      order: createQuestionDto.order,
+      required: createQuestionDto.required,
+      answer_type: {
+        id: createQuestionDto.answerFormatId,
+        name: createQuestionDto.answerFormatName,
+      },
+      answers: createQuestionDto.answers || undefined,
+    }
+    try {
+      const response = await this.fetchFromApi<ApiEcanvasserSurveyQuestion>(
+        `/survey/question`,
+        ecanvasser.apiKey,
+        {
+          method: 'POST',
+          data: payload,
+        },
+      )
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to create survey question', error)
+      throw error
+    }
```

Comment:
All of these catches in these methods talking to the Ecanvasser API should be `throw`ing `BadGatewayException`, not just whatever is being coughed up from this nebulous `fetchFromApi` method.